File: finanzas/backend/app/schemas/budget.py
from pydantic import BaseModel


class BudgetStatusSchema(BaseModel):
    category_id: str
    category_name: str
    category_slug: str
    icon: str
    color: str
    budget_amount: float
    actual_amount: float
    remaining: float
    pct_used: float


class BudgetWrite(BaseModel):
    amount: float
    month: int = 0       # 0 = default for all months
    year: int = 0        # 0 = default for all years
    currency: str = "CLP"  # CLP or USD

File: finanzas/frontend/src/api/budget.ts
import { api } from './client'
import type { BudgetStatus } from './types'

export const budgetApi = {
  list: (year: number, month: number, currency = 'CLP') =>
    api.get<BudgetStatus[]>(`/api/budget?year=${year}&month=${month}&currency=${currency}`),
  set: (categorySlug: string, amount: number, month = 0, year = 0, currency = 'CLP') =>
    api.put(`/api/budget/${categorySlug}`, { amount, month, year, currency }),
}
